tests/Crypto/tst_cryptorequests/tst_cryptorequests.cpp

```diff
@@ -392,29 +396,55 @@ void tst_cryptorequests::validateCertificateChain()
     QSKIP("TODO - certificate validation not yet implemented!");
 }
 
+void tst_cryptorequests::signVerify_data()
+{
+    QTest::addColumn<CryptoManager::Algorithm>("algorithm");
+    QTest::addColumn<CryptoManager::DigestFunction>("digestFunction");
+
+    QTest::newRow("RSA + SHA256") << CryptoManager::AlgorithmRsa << CryptoManager::DigestSha256;
+    QTest::newRow("RSA + SHA512") << CryptoManager::AlgorithmRsa << CryptoManager::DigestSha512;
+    QTest::newRow("RSA + MD5") << CryptoManager::AlgorithmRsa << CryptoManager::DigestMd5;
+    QTest::newRow("EC + SHA256") << CryptoManager::AlgorithmEc << CryptoManager::DigestSha256;
+    QTest::newRow("EC + SHA512") << CryptoManager::AlgorithmEc << CryptoManager::DigestSha512;
+}
+
+static inline KeyPairGenerationParameters *getKeyPairGenerationParameters(CryptoManager::Algorithm algorithm)
+{
+    switch (algorithm)
+    {
+    case CryptoManager::AlgorithmRsa:
+        return new RsaKeyPairGenerationParameters();
```

Comment:
why construct on the heap?

Reply:
I was under the impression that the extra data from the derived classes would be lost if I just passed this on the stack.

Reply:
It shouldn't be - and if it is, that's a bug we need to fix.  The classes should be immune to slicing etc.